components/proxy/src/main/java/com/hotels/styx/StyxServer.java

```diff
@@ -221,8 +223,8 @@ private HttpHandler2 userConfiguredHttpPipeline(Environment environment, Map<Str
         return pipelineBuilder.build();
     }
 
-    private Map<String, Service> mergeServices(Map<String, Service> configServices, Map<String, Service> additionalServices) {
-        ImmutableMap.Builder<String, Service> merged = new ImmutableMap.Builder<>();
+    private Map<String, StyxService> mergeServices(Map<String, StyxService> configServices, Map<String, StyxService> additionalServices) {
+        ImmutableMap.Builder<String, StyxService> merged = new ImmutableMap.Builder<>();
```

Comment:
The benefit of builder classes is that the methods can be chained for easy of readability. E.g.
```
 return new ImmutableMap.Builder<String, StyxService>()
                .putAll(configServices)
                .putAll(additionalServices)
                .build();
```